app/main.py

```diff
@@ -1,6 +1,19 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_store = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Any:
+        key = (args, frozenset(kwargs.items()))
```

Comment:
Using `frozenset` to create a key from `kwargs` is generally fine, but it assumes that all keyword arguments are hashable. If there's a possibility of non-hashable types being passed as keyword arguments, this could raise an error. Consider handling such cases or documenting this requirement.